l10n_it_fatturapa_out_pec/models/fatturapa_attachment_out.py

```diff
@@ -0,0 +1,183 @@
+# -*- coding: utf-8 -*-
+# Author(s): Andrea Colangelo (andreacolangelo@openforce.it)
+# Copyright © 2018 Openforce Srls Unipersonale (www.openforce.it)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+import re
+
+from lxml import etree
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+RESPONSE_MAIL_REGEX = '[A-Z]{2}[a-zA-Z0-9]{11,16}_[a-zA-Z0-9]{,5}_[A-Z]{2}_' \
+                      '[a-zA-Z0-9]{,3}'
+
+
+class FatturaPAAttachmentOut(models.Model):
+    _inherit = 'fatturapa.attachment.out'
+
+    state = fields.Selection([('ready', 'Ready to Send'),
+                              ('sent', 'Sent'),
+                              ('validated', 'Validated'),
+                              ('sender_error', 'Sender Error'),
+                              ('recipient_error', 'Recipient Error'),
+                              ('rejected', 'Rejected (PA)')],
+                             string='State',
+                             default='ready',)
+
+    last_sdi_response = fields.Text(
+        string='Last Response from Exchange System', default='No response yet')
+
+    @api.multi
+    def send_via_pec(self):
+
+        mail_message = self.env['mail.message'].create({
+            'model': self._name,
+            'res_id': self.id,
+            'subject': self.name,
+            'body': 'XML file for FatturaPA {} sent to Exchange System to the'
+                    ' email address {}.'
+            .format(
+                self.name,
+                self.env.user.company_id.email_exchange_system),
+            'attachment_ids': [(6, 0, self.ir_attachment_id.ids)],
+            'email_from': self.env.user.company_id.email_from_for_fatturaPA,
+            'mail_server_id': self.env.user.company_id.sdi_channel_id.
+            pec_server_id.id,
+        })
+
+        mail = self.env['mail.mail'].create({
+            'mail_message_id': mail_message.id,
+            'body_html': mail_message.body,
+            'email_to': self.env.user.company_id.email_exchange_system,
+        })
+
+        if mail:
+            res = mail.send()
+            if res:
+                self.state = 'sent'
+
+    @api.multi
+    def parse_pec_response(self, message_dict):
```

Comment:
Potresti aggiungere dei test con gli esempi presi da http://www.fatturapa.gov.it/export/fatturazione/it/normativa/f-3.htm?